api/classes/prerequisites.js

```diff
@@ -1,233 +1,255 @@
-var Boom = require('boom');
-var Promise = require('bluebird'); // jshint ignore:line
-
-var Users = require('../modules/users/model');
-var errors = require('./errors');
-
-var prerequisites = {};
-
-/**
- * confirmRecordExists(model[, mode, requestKey, databasekey])
- *
- * Returns a HAPI pre-req package configured to
- * to fetch all matching records of the passed `model`,
- * using data from the route parameters or the request payload
- * to build the query.
- *
- * @param {Object} model - Bookshelf model for querying
- * @param {String} mode - 'param' or 'payload' (if omitted, returns all records)
- * @param {String} requestKey - key of the param/payload
- * @param {String} databaseKey - key of the database model
- *
- * @return {Promise}
- */
-prerequisites.confirmRecordExists = function(model, config) {
-  config = config || {};
-  config.databaseKey = config.databaseKey || config.requestKey;
-
-  return {
-    assign: 'records',
-    method: function(req, reply) {
-      var queryOptions = {};
-      if (config.requestKey) {
-        queryOptions.where = {};
-
-        if (config.mode === 'param') {
-          queryOptions.where[config.databaseKey] = req.params[config.requestKey];
-        } else {
-          queryOptions.where[config.databaseKey] = req.payload[config.requestKey];
+"use strict";
+
+const Boom = require(`boom`);
+const Promise = require(`bluebird`);
+
+const Users = require(`../modules/users/model`);
+const Errors = require(`./errors`);
+
+class Prerequisites {
+  /**
+  * confirmRecordExists(model[, mode, requestKey, databasekey])
+  *
+  * Returns a HAPI pre-req package configured to
+  * to fetch all matching records of the passed `model`,
+  * using data from the route parameters or the request payload
+  * to build the query.
+  *
+  * @param {Object} model - Bookshelf model for querying
+  * @param {Object} config - Contains configuration options for the query:
+  *     mode {String} - 'param' or 'payload' (if omitted, returns all records)
+  *     requestKey {String} - key of the param/payload
+  *     databaseKey {String} - key of the database model
+  *
+  * @return {Promise} - Promise fullfilled when the record is found
+  */
+  static confirmRecordExists(model, config) {
+    config = config || {};
+    config.databaseKey = config.databaseKey || config.requestKey;
```

Comment:
does this need to be a config rewrite? (i.e. persist beyond this function). If not, let's do this with a `const` instead,
```
const configKey = config.databaseKey || config.requestKey;
const requestKey = config.requestKey;
...
      queryOptions.where[configKey] = req.params[requestKey];
...

```